lib/promscrape/discoveryutils/client.go

```diff
@@ -247,8 +247,14 @@ func (c *Client) getAPIResponseWithParamsAndClientCtx(ctx context.Context, clien
 		return nil, fmt.Errorf("cannot create request for %q: %w", requestURL, err)
 	}
 
-	c.setHTTPHeaders(req)
-	c.setHTTPProxyHeaders(req)
+	err = c.setHTTPHeaders(req)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create request for %q: %w", requestURL, err)
+	}
+	err = c.setHTTPProxyHeaders(req)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create request for %q: %w", requestURL, err)
```

Comment:
Error messages should be unique within the same context to be distinguishable.

Reply:
Fixed!